src/Plugin.php

```diff
@@ -288,6 +291,33 @@ private function saveInstalledPaths()
         return $exitCode;
     }
 
+    /**
+     * Verify that the paths which were expected to be saved, have been.
+     *
+     * @return int Exit code. 0 for success, 1 for failure.
+     */
+    private function verifySaveSuccess() {
+        $exitCode      = 1;
+        $expected      = $this->installedPaths;
+        $expectedCount = count($expected);
+
+        // Request the currently set installed paths after the save.
+        $this->loadInstalledPaths();
+
+        if ($expectedCount === 0) {
+            if (count($this->installedPaths) === 0) {
+                $exitCode = 0;
+            }
+        } else {
+            $registered = array_intersect($this->installedPaths, $expected);
+            if ($expectedCount === count($registered)) {
+                $exitCode = 0;
+            }
+        }
```

Comment:
This structure feel hard to read to me. It might be a matter of taste, but as this comparison basically boils down to this:
```
if (($a && $b) || (!$a && $c)){
    $exitCode = 0;
}
```
I would prefer something more explicit, like this:

```
        $registered = array_intersect($this->installedPaths, $expected);
        $installedCount = count($this->installedPaths)
        $registeredCount = count($registered)

        if (
            ($expectedCount === 0 && $installedCount === 0)
            || ($expectedCount > 0 && $expectedCount === $registeredCount)
        ) {
            $exitCode = 0;
        }
```

Reply:
@Potherca That would mean the relatively expensive `array_intersect()` would be run, even when it is not needed (`$expectedCount === 0`).

Reply:
True, but I don't think we need to worry about optimisations too much at this point.
Readability of the code would be of greater concern as far as I'm concerned...

Reply:
In that case, we don't need to _special case_ the `$expectedCount === 0` either.

Reply:
Good point.